pkg/server/delta/v3/server.go

```diff
@@ -118,6 +118,17 @@ func (s *server) processDelta(str stream.DeltaStream, reqCh <-chan *discovery.De
 		watch := watches.deltaWatches[typ]
 		watch.nonce = nonce
 
+		if typ == resource.EndpointType {
+			watch.state.SetClusterWarming(false)
+		}
+		if typ == resource.ClusterType {
```

Comment:
What about the cases where we get a warming listener?

Reply:
Yes! When I wrote this patch, I only had dynamic Cluster, so only ClusterType was considered. If we're to merge this patch, I'll add the rest resources need warming.

Reply:
@alecholmez Thank you for taking time reviewing this.
I just read the envoy xDS protocol document again https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#resource-warming,  and found that only `Cluster` warming needs an additional `EDS` response. No RDS response is required for `Listener` warming. So I think we only need consider about the `ClusterType` in this patch, what do you think?

<img width="919" alt="image" src="https://github.com/user-attachments/assets/af57825a-4897-4d4d-9868-2aee8f895ab4">

Reply:
Hi @lukidzi, I checked your PR. As I stated above, seems only `CDS` & `EDS` needs to be taking care of ?  Anything I missed, please help point it out.

Reply:
Hi, if both LDS and RDS are in use, you may encounter an issue when a Listener enters the warming phase and depends on RDS data to proceed. My PR only addressed the issue for CDS/EDS, so it doesn’t currently resolve the LDS/RDS dependency during the warming phase.